### DISCUSSION TITLE
Creating users question

### DISCUSSION THREAD

**Discussion by richardltc** [pocketbase/pocketbase#184] [Q&A]
const createUser = async () => {
        const { user } = await client.Users.create({
            email: email,
            password: password,
            passwordConfirm: password,
        });
        const updatedRecord = await client.Records.update(
            "profiles",
            user.profile.id,
            {
                name: "test",
            }
        );

When I run the above code, my user is being created, but I'm getting an error:

`index.svelte:22 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'profile')
    at HTMLFormElement.createUser (index.svelte:22:18)`

Any ideas what I'm doing wrong?

Richard

**Comment by ganigeorgiev** [comment]
The `client.Users.create()` should directly return the user in the response there is no key `user`, so you don't need the object destructing assignment, or in other words - remove the curly brackets around the `user` variable:

```js
const createUser = async () => {
    const user = await client.Users.create({
        email: email,
        password: password,
        passwordConfirm: password,
    });
 
    const updatedRecord = await client.Records.update(
        "profiles",
        user.profile.id,
        {
            name: "test",
        }
    );
}
```

**Reply by ganigeorgiev** [reply]
In addition, depending of the API rules of your profiles collection (from "Users > Edit profile collection" cogwheel > API rules), you may need to authenticate the user before updating its profile, because by default only admins and the user owner have access to edit the user profile, so you may need to add the following call before `client.Records.Update`:
```js
...
await client.Users.authViaEmail(email, password);

const updatedRecord = await client.Records.update(
    "profiles",
    user.profile.id,
    {
        name: "test",
    }
);
```
